### ISSUE TITLE
Tacotron2 stop sign did not work . Always equaled max decoding steps?

### ISSUE THREAD

**Issue by Hongpeng1992** [TensorSpeech/TensorflowTTS#154]

**Comment by dathudeptrai** [comment]
@Hongpeng1992 can you share ur tensorboard?, and give me some information about ur dataset setting?

**Comment by Hongpeng1992** [comment]
41 tacotron2.setup_window(win_front=3, win_back=3)
 42 
 43 #tacotron2.setup_maximum_iterations(3000)
 44 
 45 
 46 # initialize melgan model
 47 
 48 with open('./examples/multiband_melgan/conf/multiband_melgan.v1.yaml') as f:
 49     mb_melgan_config = yaml.load(f, Loader=yaml.Loader)
 50 
 51 # define model and load checkpoint
 52 mb_melgan = TFMelGANGenerator(
 53         config=MultiBandMelGANGeneratorConfig(**mb_melgan_config["generator_params"]),
 54         name="melgan",)
 55 mb_melgan._build()
 56 mb_melgan.load_weights('./examples/multiband_melgan/exp/train.multiband_melgan.v1/checkpoints/generator-1200000.h5')
 57 
 58 pqmf = TFPQMF(config=MultiBandMelGANGeneratorConfig(**mb_melgan_config["generator_params"]), name="pqmf")
 59 # inference
 60 processor = LJSpeechProcessor(None, cleaner_names="english_cleaners")
 61 input_ids = processor.text_to_sequence("Recent research at Harvard has shown meditating for as little as 8 weeks, can actually increase the grey matter in the parts of the brain responsible for emotional r    egulation, and learning.")
 62 input_ids = np.concatenate([input_ids, [len(symbols) - 1]], -1)
 63 decoder_output, mel_outputs, stop_token_prediction, alignment_history = tacotron2.inference(
 64         input_ids=tf.expand_dims(tf.convert_to_tensor(input_ids, dtype=tf.int32), 0),
 65         input_lengths=tf.convert_to_tensor([len(input_ids)], tf.int32),
 66         speaker_ids=tf.convert_to_tensor([0], dtype=tf.int32),
 67 )
 68 
 69 for i, post_mel_output in enumerate(mel_outputs):
 70         stop_token = tf.math.round(tf.nn.sigmoid(stop_token_prediction[i]))  # [T]
 71         real_length = tf.math.reduce_sum(
 72                 tf.cast(tf.math.equal(stop_token, 0.0), tf.int32), -1
 73             )
 74         post_mel_output = post_mel_output[:real_length, :]
 75         post_mel_output=tf.expand_dims(post_mel_output,0)
 76 print(post_mel_output)
 77 print("generate")
 78 print(real_length)
 79 generated_subbands = mb_melgan(post_mel_output)

**Comment by Hongpeng1992** [comment]
Hi ,  I trained the model  based on both your ljspeech pipeline and Chinese dataset.Based on the above code , I found  real_length  did always equal  the maximum_iterations.

**Comment by dathudeptrai** [comment]
@Hongpeng1992 can you share ur tensorboard ?

**Comment by Hongpeng1992** [comment]
/media/ubuntu/TOSHIBA EXT/deeptts/prediction/tensorboard.png

**Comment by Hongpeng1992** [comment]
![tensorboard](https://user-images.githubusercontent.com/41531735/88508603-cd8fdb80-d011-11ea-8575-df5159e3506f.png)

**Comment by dathudeptrai** [comment]
@Hongpeng1992 a stop loss seems ok. This problem should not be occur.

**Comment by dathudeptrai** [comment]
@Hongpeng1992 is it fixed ?

**Comment by Hongpeng1992** [comment]
No,  but it might be my problem

**Comment by dathudeptrai** [comment]
@Hongpeng1992 do not forget to add stop token to ur sentences.
